src/earthen/clj_cb/bucket.clj

```diff
@@ -56,6 +56,19 @@
             .toString)
          (document->map doc))))))
 
+(defn lookup-in
+  "Retrieves sub-document values"
+  [bucket id & rest]
+  (let [builder (.lookupIn bucket id)
+        _ (.get builder (into-array rest))
+        result (.execute builder)]
+    (apply merge-with into {} (map-indexed
+                               (fn [idx itm]
+                                 {(keyword (clojure.string/replace itm #"\[|\]" {"[" "<" "]" ">"}))
+                                  (if (contains? (supers (class (.content result idx))) com.couchbase.client.java.document.json.JsonValue)
```

Comment:
If I understood correctly, what we are trying to do here is something like:
               (if (is-json-value result idx)
                      (read-content-as-json result idx)
                      (.content result idx)......?

Reply:
Correct, it is either one of these com.couchbase.client.java.document.json.JsonArray or JsonObject backed by the JsonValue object or a primitive type, string, integer.